include/libdnf/base/goal_elements.hpp

```diff
@@ -81,25 +81,27 @@ enum class GoalProblem : uint32_t {
 enum class GoalSetting { AUTO, SET_TRUE, SET_FALSE };
 enum class GoalUsedSetting { UNUSED, USED_TRUE, USED_FALSE };
 
-struct GoalJobSettings {
+struct ResolveSpecSettings {
 public:
-    GoalJobSettings() = default;
+    bool icase{false};
+    bool with_nevra{true};
+    bool with_provides{true};
+    bool with_filenames{true};
+    std::vector<libdnf::rpm::Nevra::Form> forms{};
+};
 
+struct GoalJobSettings : public ResolveSpecSettings {
```

Comment:
I'm not sure about this inheritance from a conceptual PoV. Wouldn't it be better to just have two settings objects as an argument to the goal functions that need it?